core/src/main/java/io/questdb/cutlass/http/processors/HealthCheckProcessor.java

```diff
@@ -33,18 +33,27 @@
 
 public class HealthCheckProcessor implements HttpRequestProcessor {
 
+    private final boolean pessimisticMode;
+
+    public HealthCheckProcessor(boolean pessimisticMode) {
+        this.pessimisticMode = pessimisticMode;
+    }
+
     @Override
     public void onRequestComplete(HttpConnectionContext context) throws PeerDisconnectedException, PeerIsSlowToReadException {
         HttpChunkedResponseSocket r = context.getChunkedResponseSocket();
-        final HealthMetricsImpl metrics = context.getMetrics().health();
-        final long unhandledErrors = metrics.unhandledErrorsCount();
-        if (unhandledErrors > 0) {
-            r.status(500, "text/plain");
-            r.sendHeader();
-            r.put("Status: Unhealthy\nUnhandled errors: ");
-            r.put(unhandledErrors);
-            r.sendChunk(true);
-            return;
+
+        if (pessimisticMode) {
+            final HealthMetricsImpl metrics = context.getMetrics().health();
+            final long unhandledErrors = metrics.unhandledErrorsCount();
+            if (unhandledErrors > 0) {
+                r.status(500, "text/plain");
+                r.sendHeader();
+                r.put("Status: Unhealthy\nUnhandled errors: ");
+                r.put(unhandledErrors);
+                r.sendChunk(true);
+                return;
+            }
         }
 
         r.status(200, "text/plain");
```

Comment:
The purpose of the health check is automation, so I don't know if any human would actually read the response. We include the error count to the HTTP 500 just in case someone wants to debug an unhealthy server.

Reply:
up to you, that's something we can always add later.